apps/test.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# Skip tests if the current branch's diff-tree compared to its staging merge-base doesn't touch any files in /apps.
+# Always run tests if the current branch is 'staging'.
+
+BRANCH=$(git rev-parse --abbrev-ref HEAD)
+BRANCH_COMMIT=$(git merge-base HEAD refs/remotes/origin/staging)
+if [ ${BRANCH} == 'staging' ] || git diff-tree HEAD ${BRANCH_COMMIT} --name-only | grep -q "apps$"; then
```

Comment:
the default output from `diff-tree` only goes a single folder-level deep, so this doesn't look through every file, just the top-level files/directories from the root. still i should add the start-anchor as well to be super-clear - my intention was to return `true` and execute the inner block if a line containing "apps" was present in the output from `git diff-tree`.